### ISSUE TITLE
Add espionage features

### ISSUE THREAD

**Issue by adrianlshaw** [chorhatarahuduketuri/gc#55]
Do we want to add espionage features?

**Comment by chorhatarahuduketuri** [comment]
Yes, yes we do.

**Comment by chorhatarahuduketuri** [comment]
I think we do, but that it won't be implementable prior to implementation of visibility scopes as per #9 . They may be implementable at the same time. 

Also, what do we want espionage for? (just random info from random systems/where ships are/empire stats) or more like civ where you can spy on the whole of a city/system? or put a spy in a system, then they can jump onto passing ships, and just provide visibility? do espionage spies also do sabotage?

**Comment by adrianlshaw** [comment]
> or put a spy in a system

I don't think we should require manual actions. Let's just keep it as a simple risk/reward slider for each race you are aware of. More manual controls could be added later.

One simple thing we can make is to have a statistical chance that a spy is placed on one of the enemy ships, and the sector of where the ship is becomes visible to you. 

However, until we have a proper economy and military then there won't be much to pilfer/sabotage/learn.
